tests/partition.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2022 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import math
+
+import numpy as np
+
+import cunumeric as num
+
+
+def assert_partition(a_num, kth, axis):
+
+    # compute volume
+    shape = a_num.shape
+    volume = 1
+    for i in range(a_num.ndim):
+        volume *= shape[i]
+
+    # move axis to end and flatten other
+    sort_dim = shape[axis]
+    flattened = (int)(volume / sort_dim)
+    a_mod = a_num.swapaxes(axis, a_num.ndim - 1)
+    a_mod = a_mod.reshape((flattened, sort_dim))
+
+    for segment in range(flattened):
+        if kth > 0:
+            left_part = a_mod[segment, 0:kth]
+            max_left = left_part.max()
+            if max_left > a_mod[segment, kth]:
+                print(a_num)
+                assert False
+        if kth < sort_dim:
+            right_part = a_mod[segment, kth:sort_dim]
+            min_right = right_part.min()
+            if min_right < a_mod[segment, kth]:
+                print(a_num)
+                assert False
+
+
+def assert_argpartition(a_num, a_org, kth, axis):
+    assert a_num.dtype == np.int64
+    assert a_num.ndim == a_org.ndim
+
+    # compute volume
+    shape = a_num.shape
+    shape_org = a_org.shape
+    for i in range(a_num.ndim):
+        assert shape[i] == shape_org[i]
+
+    a_reindexed = np.take_along_axis(a_org, a_num, axis=axis)
+
+    assert_partition(a_reindexed, kth, axis)
+
+
+def test_api(a=None):
+    if a is None:
+        a = np.arange(4 * 2 * 3).reshape(4, 2, 3)
+    a_num = num.array(a)
+
+    shape = a.shape
+    volume = 1
+    for i in range(a.ndim):
+        volume *= shape[i]
+
+    # partition axes
+    for i in range(a.ndim):
+        kth = math.floor(shape[i] / 2)
+        print("partition axis " + str(i))
+        assert_partition(
+            num.partition(a_num, kth=kth, axis=i).__array__(), kth, i
```

Comment:
if you will call `__array__` for all these outputs, can't you just do it inside `assert_argpartition`?